public/app/features/explore/AddToDashboard/addToDashboard.ts

```diff
@@ -0,0 +1,44 @@
+import { DataQuery } from '@grafana/data';
+import { DashboardDataDTO } from 'app/types';
+import { getDashboardSrv } from 'app/features/dashboard/services/DashboardSrv';
+import { getBackendSrv } from '@grafana/runtime';
+import { lastValueFrom } from 'rxjs';
+
+export interface SaveToNewDashboardDTO {
+  dashboardName: string;
+  folder: { title: string; id: number };
+  queries: DataQuery[];
+  visualization: string;
+}
+
+export interface SaveToExistingDashboardDTO {
+  dashboard: string;
+  queries: DataQuery[];
+  visualization: string;
+}
+
+const createDashboardApiCall = (dashboard: DashboardDataDTO, folderId: number) => {
+  // TODO: properly type this
+  return getBackendSrv().fetch<any>({
+    url: '/api/dashboards/db/',
+    method: 'POST',
+    data: {
+      dashboard,
+      folderId,
+    },
+    showErrorAlert: false,
+  });
+};
+
+const createDashboard = (dashboardName: string, folderId: number, queries: DataQuery[], visualization: string) => {
+  const dashboard = getDashboardSrv().create({ title: dashboardName }, { folderId });
+
+  dashboard.addPanel({ targets: queries, type: visualization });
+
+  return lastValueFrom(createDashboardApiCall(dashboard.getSaveModelClone(), folderId));
+};
+
+export const addToDashboard = (data: SaveToNewDashboardDTO | SaveToExistingDashboardDTO) => {
+  // TODO: if create dashboard...
```

Comment:
i'm going to remove this together with everything related to `SaveToExistingDashboardDTO` to keep PRs smaller.